### PR TITLE
RDRP-1007 new output for intram consistencies

### PR DESCRIPTION
**Pull request by AnneONS**
## Pull Request submission

*Insert detailed bullet points about your changes here!*

*Insert any instructions to help the reviewer, e.g. "install new requirements from `requirements.txt`"*

*Let the reviewer know what data files are needed (and if applicable, where they are to be found)

#### Closes or fixes

* Detail the ticket(s) you are closing with this PR
Closes #

#### Code

- [ ] **Code runs** The code runs on my machine and/or CDSW
- [ ] **Conflicts resolved** There are no conflicts (I have performed a rebase if necessary)
- [ ] **Requirements** My/our code functions according to the requirements of the ticket
- [ ] **Dependencies** I have updated the environment yaml so it includes any new libraries I have used
- [ ] **Configuration file updated** any high level parameters that the user may interact with have been put into the config file (and imported to the script)
- [ ] **Clean Code**
    - [ ] Code is as [PEP 8]([url](https://peps.python.org/pep-0008/)) compliant as I can humanly make it
    - [ ] Code passess flake8 linting check
    - [ ] Code adheres to [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)
- [ ] **Type hints** All new functions have [type hints ](https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html)

#### Documentation

Any new code includes all the following forms of documentation:

- [ ] **Function Documentation** Docstrings within the function(s')/methods have been created
    - [ ] Includes `Args` and `returns` for all major functions
    - [ ] The docstring details data types
- [ ] **Updated Documentation**: User and/or developer working doc has been updated

#### Data
- [ ] All data needed to run this script is available in Dev/Test
- [ ] All data is excluded from this pull request
- [ ] Secrets checker pre-commit passes

#### Testing
- [ ] **Unit tests** Unit tests have been created and are passing _or a new ticket to create tests has been created_

---

# Peer Review Section

- [ ] All requirements install from (updated) `requirements.txt`
- [ ] Documentation has been created and is clear - **check the working document**
- [ ] Doctrings ([Google format](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html)) have been created and accurately describe the function's functionality
- [ ] Unit tests pass, or if not present _a new ticket to create tests has been created_
- [ ] **Code runs** The code runs on reviewer's machine and/or CDSW

#### Final approval (post-review)

The author has responded to my review and made changes to my satisfaction.
- [ ] **I recommend merging this request.**

---

### Review comments

*Insert detailed comments here!*

These might include, but not exclusively:

- bugs that need fixing (does it work as expected? and does it work with other code
  that it is likely to interact with?)
- alternative methods (could it be written more efficiently or with more clarity?)
- documentation improvements (does the documentation reflect how the code actually works?)
- additional tests that should be implemented (do the tests effectively assure that it
  works correctly?)
- code style improvements (could the code be written more clearly?)
- Do the changes represent a change in functionality so the version number should increase? Start a discussion if so.
- As a review you can generates the same outputs from running the code

Your suggestions should be tailored to the code that you are reviewing.
Be critical and clear, but not mean. Ask questions and set actions.

### CODE REVIEW

File: src/outputs/outputs_main.py [hunk 1]
```diff
@@ -119,11 +120,12 @@ def run_outputs(  # noqa: C901
...
-        output_intram_by_pg(
+        intram_tot_dict = output_intram_by_pg(
             outputs_df,
             ni_full_responses,
             pg_detailed,
             config,
+            intram_tot_dict,
             write_csv,
             run_id,
             uk_output=False,
```
Comment (JenCheshire) [inline src/outputs/outputs_main.py:131]: I can't comment on the specific line, but intram_tot_dict is missing from from the PG output args (line 148)

File: src/outputs/outputs_main.py [hunk 2]
```diff
@@ -152,10 +154,11 @@ def run_outputs(  # noqa: C901
...
         OutputMainLogger.info("Starting Intram by ITL (GB) output...")
-        output_intram_by_itl(
+        intram_tot_dict = output_intram_by_itl(
             outputs_df,
             ni_full_responses,
             config,
+            intram_tot_dict,
             write_csv,
             run_id,
         )
```
Comment (JenCheshire) [inline src/outputs/outputs_main.py:165]: As above - intram_tot_dict missing line 180

File: src/user_config.yaml [hunk 3]
```diff
@@ -34,18 +34,19 @@ global:
...
   output_intram_uk_itl: False
   output_intram_by_civil_defence: False
-  output_intram_by_sic: False
+  output_intram_by_sic: True
   output_fte_total_qa: False
   output_status_filtered: False
-  output_frozen_group: False
+  output_frozen_group: True
+  output_intram_totals: True
 hdfs_paths:
```
Comment (JenCheshire) [inline src/user_config.yaml]: Change back to False for develop

File: src/outputs/outputs_main.py (hunk 1 above)
Comment (AnneONS) [reply src/outputs/outputs_main.py:131]: the GB will be a different total to the UK, so I thought I'd leave it out. What do you think?

File: src/outputs/outputs_main.py (hunk 2 above)
Comment (AnneONS) [reply src/outputs/outputs_main.py:165]: again, not sure about including GB. We could ask Dan and add it later perhaps?

File: src/outputs/outputs_main.py (hunk 1 above)
Comment (JenCheshire) [reply src/outputs/outputs_main.py:131]: Yes, let's ask Dan (it'll need taking out of output_intram_by_pg in the meantime, as the pipeline errors)

File: src/outputs/outputs_main.py (hunk 2 above)
Comment (JenCheshire) [reply src/outputs/outputs_main.py:165]: the gb and uk itl outputs use the same main function: output_intram_by_itl, so needs to be there at the moment so it doesn't error

File: src/outputs/gb_sas.py [hunk 4]
```diff
@@ -14,6 +14,7 @@
 def output_gb_sas(
     df: pd.DataFrame,
     config: Dict[str, Any],
+    intram_tot_dict: Dict[str, int],
     write_csv: Callable,
     run_id: int,
 ):
```
Comment (JenCheshire) [inline src/outputs/gb_sas.py]: -> Dict[str, int]:

File: src/outputs/outputs_main.py (hunk 2 above)
Comment (AnneONS) [reply src/outputs/outputs_main.py:165]: silly me! Good thing you noticed!

File: src/outputs/outputs_main.py (hunk 1 above)
Comment (AnneONS) [reply src/outputs/outputs_main.py:131]: sorry, I didn't think about the errors.

**Review by JenCheshire — APPROVED** [review]
Such a helpful output!

**Pull request merged**